### DISCUSSION TITLE
[FR]s/ideas for the engine that would totally make it awesome

### DISCUSSION THREAD

**Discussion by devxan** [dotargz/novel-engine#3] [Ideas]
See my [updated comment](https://github.com/dotargz/novel-engine/discussions/3#discussioncomment-8375340)

- Run js scripts upon a choice button click/page load (for things like confetti!)
- Save system (download a file and ability to load it) beyond the current in-browser one & reset button
  - Encrypted file? decrypt on the server 💀? 
  - Save code that's super small but saved somewhere, both on client and server
- Pronouns 😨 
- Avatar/pfp photos for speakers
- Markdown support 🗿 
- Change css based on certain variables (location=hell's lobby? red chairs and fiery background!!)

**Comment by dotargz** [comment]
Thanks for the ideas! You can already set things like pronouns using the set action on any choice, and I'll definitely consider markdown support (even though you can do ``<b></b>`` and ``<i></i>`` in your text). Dynamic CSS themes on a per-page basis sounds really interesting so I'll definitely try that out. A save system is already mostly in place, and I'm not sure how to make it without using a DB and setting up accounts, but its a decent idea. Avatars for speakers would be interesting but I'm not sure where or how i would implement that right away.

I don't think I'll do JS scripts on a click because the page is reloaded immediately. Maybe someone could make a queue system for the script as a PR, but I'm not sure if I want to implement it. I really like how simple it is now.

**Comment by devxan** [comment]
Additional comments / rewording my Feature Requests:
- Run js scripts upon specific page load 
- Save file system (beyond the current in-browser one with a reset button)
  - Encrypted file? decrypt on the server 💀? 
- Per page styles 

Solved: (main goal: don't bloat the rendering engine or json object)
- ~~Pronouns 😨~~ Achievable with variables
- ~~Markdown support 🗿~~ Nvm; just find a tool that turns .md contents into .html
- ~~Avatar/pfp photos for speakers~~
  - Could make a custom element/reusable web components & include it in docs or demo
